cosmo_tester/test_suites/cluster/cfy_cluster_manager_test.py

```diff
@@ -226,6 +215,56 @@ def test_three_nodes_cluster_offline(
                      logger)
 
 
+@pytest.mark.parametrize('base_version', ['5_1_0', '5_1_1'])
+def test_three_nodes_cluster_upgrade(base_version, three_vms,
+                                     three_nodes_config_dict, test_config,
+                                     ssh_key, logger):
+    """Tests the command cfy_cluster_manager upgrade on a 3 nodes cluster."""
+    node1, node2, node3 = three_vms
+    nodes_list = [node1, node2, node3]
```

Comment:
Is there a reason to not use the same list ?

Reply:
As `nine_vms`, `three_vms` is also a generator and not a list.